src/OpenTelemetry.Api/Context/Propagation/TraceContextPropagator.cs

```diff
@@ -0,0 +1,265 @@
+// <copyright file="TraceContextPropagator.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+using OpenTelemetry.Internal;
+
+namespace OpenTelemetry.Context.Propagation
+{
+    /// <summary>
+    /// A text map propagator for W3C trace context. See https://w3c.github.io/trace-context/.
+    /// </summary>
+    public class TraceContextPropagator : TextMapPropagator
+    {
+        private const string TraceParent = "traceparent";
+        private const string TraceState = "tracestate";
+
+        private static readonly int VersionPrefixIdLength = "00-".Length;
+        private static readonly int TraceIdLength = "0af7651916cd43dd8448eb211c80319c".Length;
```

Comment:
not blocking this PR, the official name (according to the TraceContext spec) should be ParentId instead of TraceId.

Reply:
This TraceId is TraceId itself in TraceContext spec. ParentId is what we call SpanId. But spec says "Span-id" is alternate name for parentid.

Reply:
Sorry I was looking at a wrong place, meant to say `SpanIdLength` should be `ParentIdLength`.